src/PowerShellEditorServices/Language/FindDotSourcedVisitor.cs

```diff
@@ -32,15 +42,52 @@ public FindDotSourcedVisitor()
         /// or a decision to continue if it wasn't found</returns>
         public override AstVisitAction VisitCommand(CommandAst commandAst)
         {
-            if (commandAst.InvocationOperator.Equals(TokenKind.Dot) &&
-                commandAst.CommandElements[0] is StringConstantExpressionAst)
+            CommandElementAst commandElementAst = commandAst.CommandElements[0];
+            if (commandAst.InvocationOperator.Equals(TokenKind.Dot))
             {
-                // Strip any quote characters off of the string
-                string fileName = commandAst.CommandElements[0].Extent.Text.Trim('\'', '"');
-                DotSourcedFiles.Add(fileName);
+                string path;
+                switch (commandElementAst)
+                {
+                    case StringConstantExpressionAst stringConstantExpressionAst:
+                        path = stringConstantExpressionAst.Value;
+                        break;
+
+                    case ExpandableStringExpressionAst expandableStringExpressionAst:
+                        path = GetPathFromExpandableStringExpression(expandableStringExpressionAst);
+                        break;
+
+                    default:
+                        path = null;
+                        break;
+                }
+
+                if (!string.IsNullOrWhiteSpace(path))
+                {
+                    DotSourcedFiles.Add(PathUtils.NormalizePathSeparators(path));
+                }
             }
 
             return base.VisitCommand(commandAst);
         }
+
+        private string GetPathFromExpandableStringExpression(ExpandableStringExpressionAst expandableStringExpressionAst)
+        {
+            var path = expandableStringExpressionAst.Value;
+            foreach (var nestedExpression in expandableStringExpressionAst.NestedExpressions)
+            {
+                // If the string contains the variable $PSScriptRoot, we replace it with the corresponding value.
+                if (nestedExpression is VariableExpressionAst variableExpressionAst
+                    && variableExpressionAst.VariablePath.UserPath.Equals("PSScriptRoot", StringComparison.OrdinalIgnoreCase))
+                {
+                    path = path.Replace(variableExpressionAst.ToString(), _psScriptRoot);
+                }
+                else
```

Comment:
Rather than have an `else` clause, I would negate the top condition:

```csharp
if (!(nestedExpression is VariableExpressionAst variableAst
      && variableAst.VariablePath.UserPath.Equals("PSScriptRoot", StringComparison.OrdinalIgnoreCase)))
{
    return null;
}

path = path.Replace(variableAst.ToString(), _psScriptRoot);